apps/ops/serializers/adhoc.py

```diff
@@ -3,17 +3,20 @@
 from django.utils.translation import gettext_lazy as _
 from rest_framework import serializers
 
-from common.serializers.fields import ReadableHiddenField, LabeledChoiceField
+from common.serializers import WritableNestedModelSerializer
+from common.serializers.fields import ReadableHiddenField
 from common.serializers.mixin import CommonBulkModelSerializer
 from .mixin import ScopeSerializerMixin
-from ..const import Scope
 from ..models import AdHoc
+from ops.serializers import AdhocVariableSerializer
 
 
-class AdHocSerializer(ScopeSerializerMixin, CommonBulkModelSerializer):
+class AdHocSerializer(ScopeSerializerMixin, CommonBulkModelSerializer, WritableNestedModelSerializer):
     creator = ReadableHiddenField(default=serializers.CurrentUserDefault())
+    variable = AdhocVariableSerializer(many=True, required=False, allow_null=True, label=_('Variable'))
 
     class Meta:
         model = AdHoc
         read_only_field = ["id", "creator", "date_created", "date_updated", "created_by"]
-        fields = read_only_field + ["id", "name", "scope", "module", "args", "comment"]
+        fields_m2m = ['variable']
+        fields = read_only_field + fields_m2m + ["id", "name", "scope", "module", "args", "comment"]
```

Comment:
There is no significant irregularity and potential issue detected in this code snippet; however, there seems to be an inconsistency regarding whether it's being used as `AdHocSerializer` or `WritableNestedModelSerializer`. For consistency with the previous code snippets using nested models (e.g. Django REST framework), we should refer to using `ReadableHiddenField`, not `WritableNestedModelSerializer` here:

```python
-    field = ReadableHiddenField()
```

For consistency in relation to other usage within different context(s):

```python
field = ReadableHiddenField(default=serializers.CurrentUserDefault())  # Use default user if present 
```
This will align better with its expected use case and conventions.